puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman-x-develop-release.groovy

```diff
@@ -17,51 +17,59 @@ pipeline {
     stages {
         stage('Setup Environment') {
             steps {
-                dir('foreman-installer') {
-                    git(url: 'https://github.com/theforeman/foreman-installer.git', branch: 'develop')
-                    script { commit_hash = archive_git_hash() }
+                setup_nightly_build_environment(
+                    git_url: git_url,
+                    git_ref: (env.git_ref ?: git_ref),
```

Comment:
If you rerun the job (in case it fails) you could be rebuilding the wrong version. Could we error out if no git_ref was given? That's probably enough to prevent human error.

Reply:
I need to look at Jenkins to confirm, but I think it copies the same parameters if you rebuild, yes?

Reply:
If you build the job without handing in a `git_ref`, the job just assumes `develop`. Is that not desired?

Reply:
HEAD would not be desired given we want to build the last successfully tested commit. A human could enter HEAD but that should be a conscious choice which is what @ekohl is getting at.

Reply:
Okay. I've refactored these jobs to error if there is no `git_ref` given.